cronjob_scripts/get_latest_version.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+import json
+from typing import TypedDict
+import requests
+
+import semver
+
+
+def get_index_url(crate: str):
+    """
+    Packages with 1 character names are placed in a directory named 1.
+    Packages with 2 character names are placed in a directory named 2.
+    Packages with 3 character names are placed in the directory 3/{first-character} where {first-character} is the first character of the package name.
+    All other packages are stored in directories named {first-two}/{second-two} where the top directory is the first two characters of the package name, and the next subdirectory is the third and fourth characters of the package name. For example, cargo would be stored in a file named ca/rg/cargo.
+    -- https://doc.rust-lang.org/cargo/reference/registry-index.html#index-files
+    """
+    if len(crate) == 1:
+        return f"https://index.crates.io/1/{crate}"
+    elif len(crate) == 2:
+        return f"https://index.crates.io/2/{crate}"
+    elif len(crate) == 3:
+        return f"https://index.crates.io/3/{crate[0]}/{crate}"
+    else:
+        return f"https://index.crates.io/{crate[:2]}/{crate[2:4]}/{crate}"
+
+
+class CrateVersionDict(TypedDict):
+    name: str
+    vers: str
+    features: dict[str, list[str]]
+
+
+def get_latest_version(crate: str) -> CrateVersionDict:
```

Comment:
annoyingly, the sparse index doesn't calculate this for you. I think my logic is correct, and the sparse index api doesn't have a rate limit, so I'd prefer to use it if possible.

I guess we could grep for max_stable_version in the crates.io source code if we wanted to double-check, but I don't think it's the end of the world if we get it wrong.

Reply:
in theory, we could make our get_latest_version() in cargo-quickinstall rust client use the crates.io api as well, and keep the logic in sync somehow?

Reply:
Another option would be to use the crates-io dump.

When playing around with it, I discovered that it does have `max_stable_version`

Given that it's updated daily, it should be good enough.

Reply:
I would like to have so that we can `make release` and have it immediately run `make recheck` and find the latest version of cargo-quickinstall that we just published, and trigger rebuilds for all architectures.

I **think** that the sparse index protocol gives us this, but I've not tested it.